projects/novo-elements/src/elements/data-table/state/data-table-state.service.ts

```diff
@@ -12,7 +12,7 @@ export class DataTableState<T> {
   public dataLoaded = new Subject();
 
   sort: { id: string; value: string } = undefined;
-  filter: { id: string; value: string | string[] } = undefined;
+  filter: IDataTableFilter | IDataTableFilter[] = undefined;
```

Comment:
I'm not sure why these are explicitly initialized as undefined, but I do know that in order for the data-tables to filter properly these must be undefined rather than an empty object. The type change is indeed the only part related to my changes.